@here/harp-materials/lib/MapMeshMaterials.ts

```diff
@@ -706,16 +857,34 @@ export class ExtrusionFeatureMixin implements ExtrusionFeature {
      * Apply the extrusionRatio value from the parameters to the respective properties.
      */
     protected applyExtrusionParameters(params?: ExtrusionFeatureParameters) {
+        // Prepare maps for holding uniforms and defines references from the actual material.
+        linkMixinWithMaterial(this, this);
+
+        assert(this.shaderDefines !== undefined);
+        assert(this.shaderUniforms !== undefined);
+
+        // Create uniform with default value, this ensures that it is always created,
+        // so no need for checks in setters.
+        this.shaderUniforms!.extrusionRatio = new THREE.Uniform(
+            ExtrusionFeatureDefs.DEFAULT_RATIO_MAX
+        );
+
         // Apply initial parameter values.
         if (params !== undefined) {
             if (params.extrusionRatio !== undefined) {
                 this.setExtrusionRatio(params.extrusionRatio);
             }
         }
 
-        (this as any).onBeforeCompile = (shader: any) => {
-            ExtrusionFeature.onBeforeCompile(this, shader);
-        };
+        const material = this as any;
```

Comment:
Minor: We could probably get rid of this cast to any, by adding `onBeforeCompile` to the `HiddenThreeJSMaterialProperties` interface.

Reply:
Yes we may do this... although is it worth effort... Will need to be added to all implementing classes (mixins).

Reply:
Ok done!